clients/src/main/java/org/apache/kafka/common/record/FileRecords.java

```diff
@@ -298,11 +298,26 @@ public int writeTo(TransferableChannel destChannel, int offset, int length) thro
      * @return the batch's base offset, its physical position, and its size (including log overhead)
      */
     public LogOffsetPosition searchForOffsetWithSize(long targetOffset, int startingPosition) {
+        FileChannelRecordBatch prevBatch = null;
+
         for (FileChannelRecordBatch batch : batchesFrom(startingPosition)) {
-            long offset = batch.lastOffset();
-            if (offset >= targetOffset)
-                return new LogOffsetPosition(batch.baseOffset(), batch.position(), batch.sizeInBytes());
+            if (batch.baseOffset() >= targetOffset) {
```

Comment:
Could you please add comments to explain why we want to avoid exercising `lastOffset` ?